domain/test/usecases/shared_space/get_all_shared_space_members_interactor_test.dart

```diff
@@ -0,0 +1,76 @@
+// terms of the GNU Affero General Public License as published by the Free Software
+// Foundation, either version 3 of the License, or (at your option) any later version,
+// provided you comply with the Additional Terms applicable for LinShare software by
+// Linagora pursuant to Section 7 of the GNU Affero General Public License,
+// subsections (b), (c), and (e), pursuant to which you must notably (i) retain the
+// display in the interface of the “LinShare™” trademark/logo, the "Libre & Free" mention,
+// the words “You are using the Free and Open Source version of LinShare™, powered by
+// Linagora © 2009–2020. Contribute to Linshare R&D by subscribing to an Enterprise
+// offer!”. You must also retain the latter notice in all asynchronous messages such as
+// e-mails sent with the Program, (ii) retain all hypertext links between LinShare and
+// http://www.linshare.org, between linagora.com and Linagora, and (iii) refrain from
+// infringing Linagora intellectual property rights over its trademarks and commercial
+// brands. Other Additional Terms apply, see
+// <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf>
+// for more details.
+// This program is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+// FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for
+// more details.
+// You should have received a copy of the GNU Affero General Public License and its
+// applicable Additional Terms for LinShare along with this program. If not, see
+// <http://www.gnu.org/licenses/> for the GNU Affero General Public License version
+//  3 and <http://www.linshare.org/licenses/LinShare-License_AfferoGPL-v3.pdf> for
+//  the Additional Terms applicable to LinShare software.
+//
+
+import 'package:dartz/dartz.dart';
+import 'package:domain/domain.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:testshared/testshared.dart';
+
+import '../../mock/repository/shared_space/mock_shared_space_member_repository.dart';
+
+void main() {
+  group('get_all_shared_space_members_interactor', () {
+    MockSharedSpaceMemberRepository sharedSpaceMemberRepository;
+    GetAllSharedSpaceMembersInteractor _getAllSharedSpaceMembersInteractor;
+
+    setUp(() {
+      sharedSpaceMemberRepository = MockSharedSpaceMemberRepository();
+      _getAllSharedSpaceMembersInteractor = GetAllSharedSpaceMembersInteractor(sharedSpaceMemberRepository);
+    });
+
+    test('getAllSharedSpaceMembersInteractor should return success with members list', () async {
+      when(sharedSpaceMemberRepository.getMembers(sharedSpaceId1)).thenAnswer((_) async => [sharedSpaceMember1, sharedSpaceMember2]);
+
+      final result = await _getAllSharedSpaceMembersInteractor.execute(sharedSpaceId1);
+
+      final membersList = result.map((success) => (success as SharedSpaceMembersViewState).members)
```

Comment:
You should use is change because as:Ω`success is SharedSpaceMembersViewState ? success.members`

Reply:
I want to return a exact list and It'll return null if there is any problem

Reply:
I think `map((success) => success is SharedSpaceMembersViewState ? success.members : [])` is the same as mine

Reply:
If success is not an instance of SharedSpaceMembersViewState it will fire an exception while runtime